tests/test_visitors/test_ast/test_functions/test_stop_iteration.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    StopIterationInsideGeneratorViolation,
+)
+from wemake_python_styleguide.visitors.ast.functions import (
+    FunctionDefinitionVisitor,
+)
+
+stop_iteration_generator_method = """
+class CheckStopIteration():
+    def check_stop_iteration(self):
+        yield
```

Comment:
Can you please change the template generation here. I see this process as:

```python
stop_iteration_generator_function = """
def check_stop_iteration():
    {0}
    raise {1}
"""
```

Things that can be inserted as `{0}`:
- `yield`
- `yield some_parameter`
- `yield from generator()`
- `print('not a generator')` - in this case we do not raise a violation at all

Moreover, we have `mode` fixture to convert `def -> async def`, so there's no need to have explicit `async def` functions. Only functions and methods.
See an example: https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_visitors/test_ast/test_functions/test_complex_default_values.py#L32

 

```

Reply:
There is one problem here with sync and async generators as async generators do not support `yield from`.

Reply:
So it's required to test `yield from` separately for sync generators only without using `mode` fixture. Would it be fine?